braindecode/augmentation/transforms.py

```diff
@@ -0,0 +1,1000 @@
+       Learning from Heterogeneous EEG Signals with Differentiable Channel
+       Reordering. arXiv preprint arXiv:2010.13694.
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=0.2,
+        mag_range=None,
+        random_state=None
+    ):
+        super().__init__(
+            operation=channel_shuffle,
+            probability=probability,
+            magnitude=magnitude,
+            random_state=random_state
+        )
+
+
+class GaussianNoise(Transform):
+    """Randomly add white noise to all channels
+
+    Suggested e.g. in [1]_, [2]_ and [3]_
+
+    Parameters
+    ----------
+    probability : float | callable
+        Either a float between 0 and 1 defining the
+        uniform probability of applying the operation, or a `callable`
+        (modelling a non-uniform probability law for example) taking no input
+        and returning a boolean.
+    magnitude : float | None, optional
+        Float between 0 and 1 encoding the standard deviation to use for the
+        additive noise:
+        ```
+        std = magnitude * mag_range[1] + (1 - magnitude) * mag_range[0]
+        ```
+        Defaults to None (ignored).
+    mag_range : tuple of two floats | None, optional
+        Std range when set using the magnitude (see `magnitude`).
+        If omitted, the range (0, 0.2) will be used.
+    std : float, optional
+        Standard deviation to use for the additive noise. Will be ignored if
+        magnitude is not set to None. Defaults to 0.1.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Defaults to None.
+
+    References
+    ----------
+    .. [1] Wang, F., Zhong, S. H., Peng, J., Jiang, J., & Liu, Y. (2018). Data
+       augmentation for eeg-based emotion recognition with deep convolutional
+       neural networks. In International Conference on Multimedia Modeling
+       (pp. 82-93).
+    .. [2] Cheng, J. Y., Goh, H., Dogrusoz, K., Tuzel, O., & Azemi, E. (2020).
+       Subject-aware contrastive learning for biosignals. arXiv preprint
+       arXiv:2007.04871.
+    .. [3] Mohsenvand, M. N., Izadi, M. R., & Maes, P. (2020). Contrastive
+       Representation Learning for Electroencephalogram Classification. In
+       Machine Learning for Health (pp. 238-253). PMLR.
+
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=None,
+        mag_range=(0, 0.2),
+        std=0.1,
+        random_state=None
+    ):
+        # override std value when a magnitude is passed
+        if magnitude is not None:
+            min_val, max_val = mag_range
+            self.std = magnitude * max_val + (1 - magnitude) * min_val
+        else:
+            self.std = std
+
+        super().__init__(
+            operation=add_gaussian_noise,
+            probability=probability,
+            magnitude=magnitude,
+            mag_range=mag_range,
```

Comment:
Actually, the `mag_range` is not used in `Transform` so I think it would be best to remove it and simply pass the `std`. This is the same for other transform and it will avoid many of the `always ignored` for this arguments.